From 4df411aa6f42be6fe5881d7b2fb88aae9d9a56a8 Mon Sep 17 00:00:00 2001
From: mokopatches <mokopatches@openmoko.org>
Date: Fri, 25 Jul 2008 23:05:21 +0100
Subject: [PATCH] gta02-sound-bandaid.patch
 http://bugzilla.openmoko.org/cgi-bin/bugzilla/show_bug.cgi?id=1172

This patch seems to alleviate the symptoms but doesn't cure them.
Keep it to keep development going, until we have a proper solution.
---
 sound/soc/codecs/wm8753.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/sound/soc/codecs/wm8753.c b/sound/soc/codecs/wm8753.c
index fb41826..72baf3d 100644
--- a/sound/soc/codecs/wm8753.c
+++ b/sound/soc/codecs/wm8753.c
@@ -1589,6 +1589,9 @@ static int wm8753_init(struct snd_soc_device *socdev)
 	schedule_delayed_work(&codec->delayed_work,
 		msecs_to_jiffies(caps_charge));
 
+	/* Fix reg WM8753_ADCTL2 */
+	wm8753_write(codec, WM8753_ADCTL2, 0x0000);
+
 	/* set the update bits */
 	reg = wm8753_read_reg_cache(codec, WM8753_LDAC);
 	wm8753_write(codec, WM8753_LDAC, reg | 0x0100);
-- 
1.5.6.3

